From 9c488f4bf85343baa34ad3ddb2b0ad9c4ba9e09e Mon Sep 17 00:00:00 2001
From: Bachpooh <76820483+bachnxuan@users.noreply.github.com>
Date: Mon, 19 May 2025 19:57:40 +0700
Subject: [PATCH] kernel: apk_sign: add more size/hash pairs

* This will make it support:
- Official KernelSU Manager
- 5ec1cff's MKSU
- KernelSU Next
- backslashxx's MKSU
- rsuntk's MKSU
- SukiSU-Ultra

Signed-off-by: bachnxuan <bachnxuan@gmail.com>
---
 kernel/apk_sign.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/kernel/apk_sign.c b/kernel/apk_sign.c
index 87401814e..f5b51d923 100644
--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -316,5 +316,12 @@ module_param_cb(ksu_debug_manager_uid, &expected_size_ops,
 
 bool ksu_is_manager_apk(char *path)
 {
-	return check_v2_signature(path, EXPECTED_NEXT_SIZE, EXPECTED_NEXT_HASH);
-}
\ No newline at end of file
+	return (check_v2_signature(path, EXPECTED_NEXT_SIZE, EXPECTED_NEXT_HASH)
+	|| check_v2_signature(path, 0x033b, "c371061b19d8c7d7d6133c6a9bafe198fa944e50c1b31c9d8daa8d7f1fc2d2d6") // tiann/KernelSU
+	|| check_v2_signature(path, 384, "7e0c6d7278a3bb8e364e0fcba95afaf3666cf5ff3c245a3b63c8833bd0445cc4") // 5ec1cff/KernelSU
+	|| check_v2_signature(path, 0x3e6, "79e590113c4c4c0c222978e413a5faa801666957b1212a328e46c00c69821bf7") // rifsxd/KernelSU-Next
+	|| check_v2_signature(path, 0x396, "f415f4ed9435427e1fdf7f1fccd4dbc07b3d6b8751e4dbcec6f19671f427870b") // rsuntk/KernelSU
+	|| check_v2_signature(path, 0x363, "4359c171f32543394cbc23ef908c4bb94cad7c8087002ba164c8230948c21549") // backslashxx/KernelSU
+	|| check_v2_signature(path, 0x35c, "947ae944f3de4ed4c21a7e4f7953ecf351bfa2b36239da37a34111ad29993eef")  // ShirkNeko/SukiSU-Ultra
+	);
+}
